Content.Server/AI/Utility/Actions/UtilityAction.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Collections.Generic;
+using Content.Server.AI.Operators;
+using Content.Server.AI.Utility.Considerations;
+using Content.Server.AI.WorldState;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Utility;
+
+namespace Content.Server.AI.Utility.Actions
+{
+    /// <summary>
+    /// The same DSE can be used across multiple actions.
+    /// </summary>
+    public abstract class UtilityAction : IAiUtility
+    {
+        /// <summary>
+        /// If we're trying to find a new action can we replace a currently running one with one of the same type.
+        /// e.g. If you're already wandering you don't want to replace it with a different wander.
+        /// </summary>
+        public virtual bool CanOverride => false;
+
+        /// <summary>
+        /// This is used to sort actions; if there's a top-tier action available we won't bother checking the lower tiers.
+        /// Threshold doesn't necessarily mean we'll do an action at a higher threshold;
+        /// if it's really un-optimal (i.e. low score) then we'll also check lower tiers
+        /// </summary>
+        /// Guidelines:
+        // Idle should just be flavor-stuff if we reeaaalllyyy have nothing to do
+        // Idle = 1
+        // Normal = 5
+        // Needs = 10
+        // Combat prep (e.g. grabbing weapons) = 20
+        // Combat = 30
+        // Danger (e.g. dodging a grenade) = 50
```

Comment:
Might be worth defining these as constants.

Reply:
That's a good idea; I could use them as constants and then each action could be +/- from the baseline, thanks CI.